src/main/java/org/ibp/api/java/impl/middleware/study/StudyTreeServiceImpl.java

```diff
@@ -0,0 +1,128 @@
+import org.generationcp.commons.pojo.treeview.TreeNode;
+import org.generationcp.commons.util.TreeViewUtil;
+import org.generationcp.middleware.api.program.ProgramDTO;
+import org.generationcp.middleware.domain.dms.Reference;
+import org.generationcp.middleware.manager.api.StudyDataManager;
+import org.generationcp.middleware.pojos.dms.DmsProject;
+import org.ibp.api.exception.ApiRequestValidationException;
+import org.ibp.api.exception.ResourceNotFoundException;
+import org.ibp.api.java.impl.middleware.common.validator.ProgramValidator;
+import org.ibp.api.java.impl.middleware.study.validator.StudyTreeValidator;
+import org.ibp.api.java.study.StudyTreeService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.MapBindingResult;
+
+import javax.annotation.Resource;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+@Service
+@Transactional
+public class StudyTreeServiceImpl implements StudyTreeService {
+
+	@Resource
+	private StudyTreeValidator studyTreeValidator;
+
+	@Resource
+	public ProgramValidator programValidator;
+
+	@Resource
+	private StudyDataManager studyDataManager;
+
+	@Resource
+	private org.generationcp.middleware.service.api.study.StudyTreeService studyTreeService;
+
+	@Override
+	public Integer createStudyTreeFolder(final String cropName, final String programUUID, final Integer parentId, final String folderName) {
+		this.studyTreeValidator.validateFolderName(folderName);
+		this.studyTreeValidator.validateFolderId(parentId, programUUID);
+		this.validateProgram(cropName, programUUID);
+		this.studyTreeValidator.validateNotSameFolderNameInParent(folderName, parentId, programUUID);
+
+		return this.studyTreeService.createStudyTreeFolder(parentId, folderName, programUUID);
+	}
+
+	@Override
+	public Integer updateStudyTreeFolder(final String cropName, final String programUUID, final Integer parentId, final String newFolderName) {
+		this.studyTreeValidator.validateFolderName(newFolderName);
+		final DmsProject folder = this.studyTreeValidator.validateFolderId(parentId, programUUID);
+
+		//Preventing edition using the same folder name
+		if (newFolderName.equalsIgnoreCase(folder.getName())) {
+			return folder.getProjectId();
+		}
+
+		this.validateProgram(cropName, programUUID);
+		this.studyTreeValidator.validateNotSameFolderNameInParent(newFolderName, folder.getParent().getProjectId(), programUUID);
+
+		return this.studyTreeService.updateStudyTreeFolder(parentId, newFolderName);
+	}
+
+	@Override
+	public void deleteStudyFolder(final String cropName, final String programUUID, final Integer folderId) {
+		this.studyTreeValidator.validateFolderId(folderId, programUUID);
+		this.validateProgram(cropName, programUUID);
+		this.studyTreeValidator.validateFolderHasNoChildren(folderId, "study.folder.delete.has.child", programUUID);
+
+		// TODO: should only the owner be able to delete the folder?
```

Comment:
@lik-83 not sure how it was working in the previous module

Reply:
Currently, any user can delete any folder